src/workforce-management/Controllers/TrainingProgramController.cs

```diff
@@ -34,9 +37,59 @@ public TrainingProgramController(BangazonContext ctx)
         }
 
         /**
-         * Purpose: Creates a View list all Training Programs currently in the database
+         * Purpose: Creates a View of the Training Program Form and populates the employees list
+         * Return:
+         *      Create Training Program Form View
+         */
+
+        [HttpGet]
+        public IActionResult Add()
+        {
+            var model = new TrainingProgramAdd();
+            model.Employees = context.Employee
+                .OrderBy(l => l.LastName)
+                .AsEnumerable()
+                .Select(li => new SelectListItem
+                {
+                    Text = li.LastName,
+                    Value = li.EmployeeId.ToString()
+                });
+            return View(model);
+        }
+        /**
+         * Purpose: Check if the form is valid, add a new training program to database, and add attendees to the database
+         * Arguments:
+         *      TrainingProgramAdd trainingProgramAdd - TrainingProgramAdd View Model containing fields to populate the dropdownlist and receive the array of EmployeeIds. 
+         */
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Add(TrainingProgramAdd trainingProgramAdd)
+        {
+            if (ModelState.IsValid)
+            {
+                TrainingProgram newTrainingProgram = trainingProgramAdd.NewTrainingProgram;
+
+                context.Add(newTrainingProgram);
+                await context.SaveChangesAsync();
+
+                if (trainingProgramAdd.EmployeeIds.Count() >= 0)
+                {
+                    foreach (int employee in trainingProgramAdd.EmployeeIds)
+                    {
+                        context.Attendee.Add(new Bangazon.Models.Attendee { EmployeeId = employee,  ProgramId = newTrainingProgram.TrainingProgramId });
+                    }
+                }
+
+                await context.SaveChangesAsync();
+                return RedirectToAction("Detail", new RouteValueDictionary(new { controller = "TrainingProgram", action = "Detail", Id = newTrainingProgram.TrainingProgramId}) );
+            }
+            return RedirectToAction("Index", "TrainingProgram");
```

Comment:
If Model State is invalid, we are just returning them to the TrainingProgram/Index view?